### PR TITLE
Merge unit tests part3

### PR DESCRIPTION
**Pull request by OHerasym**
### PR provides following changes:
-  [x] Fix connection handler tests
- [x] Fix hmi_message_handler unit tests
- [x] Fix telemetry_monitor unit tests
- [x] Fix media_manager unit tests
- [x] Fix resumption unit tests

Correct tests according to code changes and make tests buildable

> Tests builds on LINUX&WINDOWS&QT platforms

Almost all connection handler tests disabled : (_thread qt dcheck fails_)

This PR contains changes and review notes fixes from [previous PR](https://github.com/OHerasym/sdl_core/pull/2).

### CODE REVIEW

File: src/components/connection_handler/test/heart_beat_monitor_test.cc [hunk 1]
```diff
@@ -44,6 +44,11 @@ const int32_t MICROSECONDS_IN_MILLISECONDS = 1000;
 const int32_t MICROSECONDS_IN_SECOND = 1000 * 1000;
 }
 
+#if defined(OS_WINDOWS)
+#include <winsock2.h>
+#define usleep(...) Sleep(__VA_ARGS__ * MICROSECONDS_IN_MILLISECONDS)
```
Comment (anosach-luxoft) [inline src/components/connection_handler/test/heart_beat_monitor_test.cc]: @OHerasym remove it. We have `sleep` method in thread

File: src/components/hmi_message_handler/test/CMakeLists.txt [hunk 2]
```diff
@@ -57,6 +57,11 @@ if(${QT_HMI})
     ${COMPONENTS_DIR}/hmi_message_handler/test/dbus_message_adapter_test.cc
   )
 endif()
+
 create_test("hmi_message_handler_test" "${SOURCES}" "${LIBRARIES}")
 
+if(MSVC)
+  target_link_libraries("hmi_message_handler_test" ws2_32)
```
Comment (anosach-luxoft) [inline src/components/hmi_message_handler/test/CMakeLists.txt:64]: @OHerasym what is `ws2_32`?

File: src/components/media_manager/test/CMakeLists.txt [hunk 3]
```diff
@@ -69,9 +68,27 @@ if(EXTENDED_MEDIA_MODE)
...
+  link_directories(${CMAKE_SOURCE_DIR}/build/openssl_win_x64/lib/)
+endif()
+
 create_test("media_manager_test" "${SOURCES}" "${LIBRARIES}")
 
-if(ENABLE_LOG)
+if(MSVC)
+  target_link_libraries("media_manager_test" ws2_32)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /FORCE" )
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /IGNORE:4099" )
```
Comment (anosach-luxoft) [inline src/components/media_manager/test/CMakeLists.txt:87]: @OHerasym why are we going to ignore 4099 warns?

File: src/components/hmi_message_handler/test/CMakeLists.txt (hunk 2 above)
Comment (OHerasym) [inline src/components/hmi_message_handler/test/CMakeLists.txt:64]: @anosach-luxoft , lib for win sockets , to resolve link issues

File: src/components/media_manager/test/CMakeLists.txt (hunk 3 above)
Comment (OHerasym) [inline src/components/media_manager/test/CMakeLists.txt:87]: @anosach-luxoft , sometimes VS cant find *.pdb file, in this case .exe file will be not created

File: src/components/connection_handler/test/heart_beat_monitor_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/connection_handler/test/heart_beat_monitor_test.cc]: @anosach-luxoft  , Removed

File: src/components/hmi_message_handler/test/CMakeLists.txt (hunk 2 above)
Comment (anosach-luxoft) [inline src/components/hmi_message_handler/test/CMakeLists.txt:64]: @OHerasym so why `MSVC`? Maybe `WIN_NATIVE`?

File: src/components/media_manager/test/CMakeLists.txt (hunk 3 above)
Comment (anosach-luxoft) [inline src/components/media_manager/test/CMakeLists.txt:87]: @OHerasym warnings can not disable exe file creating. Is `4099` warnings class or error?

Comment (OHerasym) [inline src/components/media_manager/test/CMakeLists.txt:87]: @anosach-luxoft , yeap, its warning, but next error is 

> all warnings being treated as errors

File: src/components/hmi_message_handler/test/CMakeLists.txt (hunk 2 above)
Comment (OHerasym) [inline src/components/hmi_message_handler/test/CMakeLists.txt:64]: @anosach-luxoft , Changed & Rebased

**Pull request merged**